addons/spike/functions/fnc_onFired.sqf

```diff
@@ -0,0 +1,89 @@
+#include "script_component.hpp"
+/*
+ * Author: Brandon (TCVM)
+ * Initialises SPIKE camera
+ *
+ * Arguments:
+ * 0: Guidance Arg Array <ARRAY>
+ * 1: PFID <NUMBER>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [[], 0] call ace_missileguidance_fnc_guidancePFH;
+ *
+ * Public: No
+ */
+params ["_shooter","_weapon","","_mode","_ammo","","_projectile"];
+
+private _missileGuidanceConfig = (configOf _projectile) >> "ace_missileguidance";
+// Setup camera array
+private _cameraConfig = _missileGuidanceConfig >> "camera";
+private _cameraArray = [false];
+if (!(_cameraConfig isEqualTo configNull) && { (getNumber (_cameraConfig >> "enabled")) == 1 }) then {
+    _cameraArray set [0, true];
+    _cameraArray set [1, getArray (_cameraConfig >> "fovLevels")];
+    _cameraArray set [2, getNumber (_cameraConfig >> "initialFOV")];
+    
+    _cameraArray set [3, getArray (_cameraConfig >> "enabledThermalTypes")];
+    _cameraArray set [4, getText (_cameraConfig >> "initialThermalType")];
+    
+    _cameraArray set [5, (getNumber (_cameraConfig >> "switchOnFire")) == 1];
+    
+    _cameraArray set [6, getNumber (_cameraConfig >> "lerpFOV")];
+    _cameraArray set [7, getNumber (_cameraConfig >> "fovChangeTime")];
+    
+    _cameraArray set [8, [[0, 0, 0], [0, 0, 0], [0, 0, 0], false, false]]; // camera view data. [look direction, ground pos, point pos, moving camera x, moving camera y]
+    
+    _cameraArray set [9, [
+        getNumber (_cameraConfig >> "gimbal" >> "enabled") == 1,
```

Comment:
I wonder if the repeating lines like this could use the `_cameraConfig >> "gimbal"` part being stored in a var.